File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SideDrawerGettingStartedModalController.swift
//
//  SideDrawerGettingStartedModalController.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.

class SideDrawerGettingStartedModalController: UIViewController, TKSideDrawerDelegate {

    
    override func viewDidLoad() {
        super.viewDidLoad()
                
        let backgroundView = UIImageView(frame: self.view.bounds)
        backgroundView.image = UIImage(named: "sdk-examples-bg")
        self.view.addSubview(backgroundView)
        
        let navigationBar = UINavigationBar(frame: CGRectMake(0, 0, self.view.frame.size.width, 64))
        let navigationItem = UINavigationItem(title: "Getting Started")
        let doneButton = UIBarButtonItem(title: "Close", style: UIBarButtonItemStyle.Done, target: self, action: "dismiss")
        let showSideDrawerButton = UIBarButtonItem(image: UIImage(named: "menu"), style: UIBarButtonItemStyle.Plain, target: self, action: "showSideDrawer")
        navigationItem.rightBarButtonItem = doneButton
        navigationItem.leftBarButtonItem = showSideDrawerButton
        navigationBar.items = [navigationItem]
        self.view.addSubview(navigationBar)
        
        let sideDrawer = self.sideDrawer
        sideDrawer.delegate = self
        sideDrawer.style.headerHeight = 64
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))

        var section = sideDrawer.addSectionWithTitle("Primary")
        section.addItemWithTitle("Social")
        section.addItemWithTitle("Promotions")
        
        section = sideDrawer.addSectionWithTitle("Label")
        section.addItemWithTitle("Important")
        section.addItemWithTitle("Starred")
        section.addItemWithTitle("Sent Mail")
        section.addItemWithTitle("Drafts")
    }
    
    func dismiss() {
        self.presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
    }
    
    func showSideDrawer() {
        self.sideDrawer.show()
    }
    
    func dismissSideDrawer() {
        self.sideDrawer.dismiss()
    }
    
    func sideDrawer(sideDrawer: TKSideDrawer!, updateVisualsForSection sectionIndex: Int) {
        let section = sideDrawer.sections()[sectionIndex] as TKSideDrawerSection
        section.style.contentInsets = UIEdgeInsetsMake(0, -15, 0, 0)
    }
    
    func sideDrawer(sideDrawer: TKSideDrawer!, updateVisualsForItem item: Int, inSection section: Int) {
        let currentItem = (sideDrawer.sections()[section] as TKSideDrawerSection).items()[item] as TKSideDrawerItem
        currentItem.style.contentInsets = UIEdgeInsetsMake(0, -10, 0, 0)
        currentItem.style.separatorColor = TKSolidFill(color: UIColor.clearColor())
    }
}

File: TelerikUIExamplesInCSharp/Examples/SideDrawer/GettingStartedExample/SideDrawerGettingStartedModalController.cs
using System;

using ObjCRuntime;
using Foundation;
using UIKit;
using CoreAnimation;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class SideDrawerGettingStartedModalController : UIViewController
	{
		private TKSideDrawerSection primarySection;
		private TKSideDrawerSection labelsSection;

		public TKSideDrawer SideDrawer {
			get;
			set;
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			UIImageView backgroundView = new UIImageView (this.View.Bounds);
			backgroundView.Image = new UIImage ("sdk-examples-bg.png");
			this.View.AddSubview (backgroundView);

			UINavigationBar navBar = new UINavigationBar (new CGRect (0, 0, this.View.Frame.Size.Width, 64));
			UINavigationItem navItem = new UINavigationItem ("SideDrawer Examples");
			UIBarButtonItem closeButton = new UIBarButtonItem ("Close", UIBarButtonItemStyle.Done, this, new Selector ("Dismiss"));
			UIBarButtonItem showSideDrawer = new UIBarButtonItem (new UIImage ("menu.png"), UIBarButtonItemStyle.Plain, this, new Selector ("ShowSideDrawer"));
			navItem.RightBarButtonItem = closeButton;
			navItem.LeftBarButtonItem = showSideDrawer;
			navBar.Items = new UINavigationItem[] { navItem };
			this.View.AddSubview (navBar);

			primarySection = new TKSideDrawerSection ("Primary");
			primarySection.AddItem ("Social");
			primarySection.AddItem ("Promotions");

			labelsSection = new TKSideDrawerSection ("Labels");
			labelsSection.AddItem ("Important");
			labelsSection.AddItem ("Starred");
			labelsSection.AddItem ("Sent Mail");
			labelsSection.AddItem ("Drafts");

			SideDrawer = TKSideDrawer.FindSideDrawer (this);
			SideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector ("DismissSideDrawer"));
			SideDrawer.AddSection (primarySection);
			SideDrawer.AddSection (labelsSection);
			SideDrawer.Delegate = new SideDrawerDelegate();
			SideDrawer.Style.HeaderHeight = 64;
		}

		[Export ("Dismiss")]
		public void Dismiss()
		{
			this.PresentingViewController.DismissViewController(true, null);
		}

		[Export ("ShowSideDrawer")]
		public void ShowSideDrawer()
		{
			SideDrawer.Show();
		}

		[Export ("DismissSideDrawer")]
		public void DismissSideDrawer()
		{
			SideDrawer.Dismiss();
		}

		class SideDrawerDelegate : TKSideDrawerDelegate
		{
			public override void UpdateVisualsForSection (TKSideDrawer sideDrawer, int sectionIndex)
			{
				TKSideDrawerSection section = sideDrawer.Sections[sectionIndex];
				section.Style.ContentInsets = new UIEdgeInsets (0, -15, 0, 0);
			}

			public override void UpdateVisualsForItem (TKSideDrawer sideDrawer, int itemIndex, int sectionIndex)
			{
				TKSideDrawerItem item = sideDrawer.Sections[sectionIndex].Items[itemIndex];
				item.Style.ContentInsets = new UIEdgeInsets (0, -5, 0, 0);
				item.Style.SeparatorColor = new TKSolidFill (UIColor.Clear);
			}
		}
	}
}

